### PR TITLE
crimson: various fmt related changes and fixes.

### PR DESCRIPTION
**Pull request by tchaikov**
these are the leftovers of #49000

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/onode_manager/staged-fltree/node.cc [hunk 1]
```diff
@@ -17,6 +17,14 @@
 
 SET_SUBSYS(seastore_onode);
 
+namespace fmt {
+
+template <typename T> auto ptr(const ::boost::intrusive_ptr<T>& p) -> const void* {
```
Comment (rzarzynski) [inline src/crimson/os/seastore/onode_manager/staged-fltree/node.cc:22]: Is there a benefit from `auto ... -> const void*`?

Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/node.cc:22]: what exactly are you referring? the trailing return type? i am following https://github.com/fmtlib/fmt/blob/7df30f91aee5444a733cec0b911d21cebdeb62ae/include/fmt/format.h#L3887-L3897 . if this is not encouraged, i will use the old-style `const void* ptr(...)`. in this case, no benefit in particular. but like the C++ Code Guidelines explains in [ `{}` ](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list). this form is more expressive, so might be advisable to just use it everywhere.

Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/node.cc:22]: guess that's why clang-tidy has this check, see https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-trailing-return-type.html

**Review by rzarzynski — APPROVED** [review]

File: src/crimson/os/seastore/onode_manager/staged-fltree/node.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/node.cc]: changed it to the old style. yeah, it increases the mental load of human readers who are not used to the trailing return type. but i think it's all about the custom of how people read/parse/understand/the code .  one can also read it like "a function named foo which takes x and y and returns bar"

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**